assets/app/view/game/button/buy_share.rb

```diff
@@ -16,6 +16,8 @@ class BuyShare < Snabberb::Component
         needs :prefix, default: 'Buy'
         needs :source, default: 'Market'
         needs :action, default: Engine::Action::BuyShares
+        needs :purchase_for, default: nil
```

Comment:
why is this needed? is there another way?

Reply:
This is for adding a button to buy a share for another player. Using it to implement the Union Bank, not another way that I know of since it's a player action that needs to be counted as their buy, but they purchase a share for another player.